### PR TITLE
Move classes to presto-common

### PR DESCRIPTION
**Pull request by jainxrohit**
We want to create abstraction for QueryPreparer so that we can support additional query preparer.  To help in this, we are moving some classes to presto-common where they actually belong.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineSemiJoinDistributionType.java [hunk 1]
```diff
@@ -48,7 +48,6 @@
 import static com.facebook.presto.SystemSessionProperties.getJoinMaxBroadcastTableSize;
 import static com.facebook.presto.SystemSessionProperties.isSizeBasedJoinDistributionTypeEnabled;
 import static com.facebook.presto.cost.CostCalculatorWithEstimatedExchanges.calculateJoinCostWithoutOutput;
-import static com.facebook.presto.sql.analyzer.FeaturesConfig.JoinDistributionType.AUTOMATIC;
```
Comment (rschlussel) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineSemiJoinDistributionType.java:51]: unrelated change

**Review by rschlussel — APPROVED** [review]

**Comment by rschlussel** [comment]
should we have a release note so people depending on the SPI know that this changed?

**Comment by jainxrohit** [comment]
> TestGeometrySerialization

These tests work in trunk.  Are they failing for you in local?

**Comment by elharo** [comment]
Locally, though I'm not sure I've ever been able to get all the tests to pass on the same run.
